test/upgradeable/behavior/nestable.ts

```diff
@@ -0,0 +1,1093 @@
+import { expect } from 'chai';
+import { ethers, upgrades } from 'hardhat';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import { BigNumber, Contract } from 'ethers';
+import { bn, ADDRESS_ZERO } from '../../utils';
+import { IERC165, IERC721, IERC6059, IOtherInterface } from '../../interfaces';
+import { upgradeable } from '../../../typechain-types/contracts';
+
+async function shouldBehaveLikeNestable(
```

Comment:
This expects contracts to be available at `this.something`, so you can reuse the existing behavior.